java/mykola-melnyk/src/com/zoolatech/lecture4/tasks/_1/Container.java

```diff
@@ -0,0 +1,33 @@
+package com.zoolatech.lecture4.tasks._1;
+
+public class Container {
+    private int value;
+
+    public Container (int value) {
+        this.value = value;
+    }
+
+    public int getValue() {
+        return value;
+    }
+
+    public void setValue(int value) {
+        this.value = value;
+    }
+
+    class ContainerUpdatesListener {
+        private int firstValue;
+
+        public ContainerUpdatesListener() {
+            Container.this.value = firstValue;
```

Comment:
it should be opposite: you need to assign the `value` to the `firstValue`; note: since variable names are different you don't need to use a container's class name to access its variable

Reply:
Done